### PR TITLE
Added page zooming option

### LINKED ISSUE #385: Scaling dashboard
**What would you like to be added**:
Is there a way to scale the dashboard? We want to render a dashboard in 1920x1080 but have the font bigger (of the titles etc). Sames as if you would press Ctrl+ in grafana on a dashboard. We found the scale option but it just seems to affect the resolution and not make the actual content bigger.

**Why is this needed**:
We want to display grafana dashboards on LFDs and need them to be bigger so they can be readable from a distance.

### PR DESCRIPTION
**Pull request by kaffarell**
Fixes #385
Added option for page zoom level. Allows dashboards to be zoomed (for bigger titles, etc.)

### CODE REVIEW

**Comment by kaffarell** [comment]
The failing check looks like an issue in the pipeline.

**Comment by kaffarell** [comment]
@AgnesToulet @joanlopez :
```
The agent is unable to authenticate with the platform without a valid token.
Please ensure you have run `srcclr activate` or have set the SRCCLR_API_TOKEN environment variable.

```

**Comment by joanlopez** [comment]
> @AgnesToulet @joanlopez :
> 
> ```
> The agent is unable to authenticate with the platform without a valid token.
> Please ensure you have run `srcclr activate` or have set the SRCCLR_API_TOKEN environment variable.
> ```

Hey @kaffarell,

Thanks for taking care!

I'm aware of such fails for "external" pull requests, I'm trying to get some engagement internally from the Release engineering teams to try to solve that issue.

Hope to reach you back sooner than later.

**Comment by kaffarell** [comment]
Hey @joanlopez any updates on this? Could you in the meantime trigger the pipeline from your account?

**Comment by kaffarell** [comment]
Hey @joanlopez any updates on this?

**Comment by joanlopez** [comment]
> Hey @joanlopez any updates on this?

Hey @kaffarell, I don't think we'll have a long-term solution soon, so meanwhile I decided to [disable those scans](https://github.com/grafana/grafana-image-renderer/pull/390), which I think is fair and would help to get your pull request merged.

Thanks for your patience! :)

**Comment by kaffarell** [comment]
Hey @joanlopez how is it looking?

**Comment by joanlopez** [comment]
Hey @kaffarell,

The aforementioned pull request has been merged, could you try to pull the latest changes from `grafana:master` and see if the pipeline succeeds now, please?

Thanks!

**Comment by kaffarell** [comment]
@joanlopez there seems to be something wrong with the pipeline...

**Comment by joanlopez** [comment]
> @joanlopez there seems to be something wrong with the pipeline...

I triggered it manually and it worked 🎉 

Planning to do a final review (and hopefully merge it!) along the week!

Thanks again for your patience and support!

**Comment by kaffarell** [comment]
Thanks a lot @joanlopez!

**Comment by kaffarell** [comment]
@joanlopez can you add a reviewer?

File: src/browser/browser.ts [hunk 1]
```diff
@@ -153,6 +153,7 @@ export class Browser {
       await client.send('Network.emulateNetworkConditions', this.config.networkConditions);
     }
 
+
```
Comment (joanlopez) [inline src/browser/browser.ts:156]:
```suggestion
```

Comment (joanlopez) [reply src/browser/browser.ts:156]: Nit

File: src/plugin/v2/grpc_plugin.ts [hunk 2]
```diff
@@ -295,6 +295,10 @@ const populateConfigFromEnv = (config: PluginConfig) => {
     config.rendering.maxDeviceScaleFactor = parseFloat(env['GF_PLUGIN_RENDERING_VIEWPORT_MAX_DEVICE_SCALE_FACTOR'] as string);
   }
 
+  if (env['GF_PLUGIN_RENDERING_VIEWPORT_PAGE_ZOOM_LEVEL']) {
+    config.rendering.pageZoomLevel = parseFloat(env['GF_PLUGIN_RENDERING_VIEWPORT_PAGE_ZOOM_LEVEL'] as string);
```
Comment (joanlopez) [inline src/plugin/v2/grpc_plugin.ts:299]: I think you missed the equivalent of this line but on `src/app.ts`.

File: src/browser/browser.ts [hunk 3]
```diff
@@ -546,4 +548,11 @@ export class Browser {
   logPageClosed = () => {
     this.log.debug('Browser page closed');
   };
+
+  private async setPageZoomLevel(page: puppeteer.Page, zoomLevel: number) {
+    this.log.debug('Setting zoom level to: ' + zoomLevel);
```
Comment (joanlopez) [inline src/browser/browser.ts:553]:
```suggestion
    this.log.debug('Setting zoom level', 'zoomLevel', zoomLevel);
```

**Comment by joanlopez** [comment]
> @joanlopez can you add a reviewer?

Just reviewed it, I left few comments that I hope make sense to you and pretty easy to fix.

Thanks!

**Comment by joanlopez** [comment]
Btw, have you seen [the issue you mentioned from Puppeteer's repo](https://github.com/puppeteer/puppeteer/issues/4419) has been closed? Any thoughts on that? 🤔

**Comment by kaffarell** [comment]
I tried some of the options in the thread from puppeteer but they didn't work for me at the time. Maybe we could evaluate later on and check if some of these https://github.com/puppeteer/puppeteer/issues/4419 fixes work. This solution is the only one that worked for me.

File: src/app.ts [hunk 4]
```diff
@@ -152,6 +152,10 @@ function populateServiceConfigFromEnv(config: ServiceConfig, env: NodeJS.Process
     config.rendering.dumpio = env['RENDERING_DUMPIO'] === 'true';
   }
 
+  if (env['VIEWPORT_PAGE_ZOOM_LEVEL']) {
+    config.rendering.pageZoomLevel = parseFloat(env['GF_PLUGIN_RENDERING_VIEWPORT_PAGE_ZOOM_LEVEL'] as string);
+  }
```
[the remark this answers is not in the document]
Comment (joanlopez) [reply src/app.ts:157]: I think we should follow the convention of the `RENDERING_` prefix, also looks like you forgot to rename the variable name within the call to `parseFloat` -- no worries!

[the remark this answers is not in the document]
Comment (kaffarell) [reply src/app.ts]: ah, yes. Thanks for the heads up!

**Review by joanlopez — COMMENTED** [review]
LGTM! 

Thanks for the contribution and commitment @kaffarell! :)

**Review by joanlopez — APPROVED** [review]

File: src/browser/browser.ts [hunk 5]
```diff
@@ -546,4 +547,11 @@ export class Browser {
   logPageClosed = () => {
     this.log.debug('Browser page closed');
   };
+
+  private async setPageZoomLevel(page: puppeteer.Page, zoomLevel: number) {
+    this.log.debug('Setting zoom level', 'zoomLevel', zoomLevel);
```
Comment (joanlopez) [inline src/browser/browser.ts:552]:
```suggestion
    if (this.config.verboseLogging) {
      this.log.debug('Setting zoom level', 'zoomLevel', zoomLevel);
    }
```

**Comment by kaffarell** [comment]
Perfect, thanks @joanlopez !

**Pull request merged**